trieste/acquisition/sampler.py

```diff
@@ -329,6 +330,61 @@ def sample(self, at: TensorType, *, jitter: float = DEFAULTS.JITTER) -> TensorTy
         return mean[..., None, :, :] + tf.transpose(variance_contribution, new_order)
 
 
+class DeepGaussianProcessSampler(Sampler):
+    r"""
+    This sampler employs the *reparameterization trick* to approximate samples from a
+    :class:`DeepGaussianProcess`\ 's predictive distribution. This sampler is essentially an
+    extension of :class:`IndependentReparametrizationSampler` for use with DGP models.
+    """
+
+    def __init__(self, sample_size: int, model: ProbabilisticModel):
```

Comment:
shouldn't model class here be `DeepGaussianProcess`?

Reply:
Wouldn't this cause a typing error because we need to init the super class `Sampler`, which takes `ProbabilisticModel`?

Reply:
`DeepGaussianProcess` is a subclass of `ProbabilisticModel` right? hence, it shouldn't because it inherits that class as well

Reply:
Oh of course